StudentDriver/StudentDriver/Services/ServiceController.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using StudentDriver.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OAuth;
+using OAuthAccess;
+using StudentDriver.Helpers;
+using Xamarin.Auth;
+using Xamarin.Auth.Presenters;
+
+namespace StudentDriver.Services
+{
+    public class ServiceController
+    {
+        private static HttpClient _client;
+        private static ServiceController _instance;
+        private readonly OAuthController _oAuthController;
+        private readonly DatabaseController _databaseController;
+
+
+        public static ServiceController Instance => _instance ?? (_instance = new ServiceController());
+
+
+        private ServiceController()
+        {
+            _client = new HttpClient();
+            _oAuthController = new OAuthController();
+            _databaseController = new DatabaseController();
+        }
+
+        //public async Task<UserStats> GetStudentStats(int id)
+        //{
+        //    var requestUri = GenerateRequestUri(_apiBaseUrl, "students", new Dictionary<string, string>() {{"userId", id.ToString()}});
+        //    var response = await _client.GetAsync(requestUri);
+        //    var json = response.Content.ReadAsStringAsync().Result;
+        //    var userStats = JsonConvert.DeserializeObject<UserStats>(json);
+        //    return userStats;
+        //}
+
+        //public async Task<bool> PostUnsyncDrivingSessions(List<UnsyncDrive> unsyncDrives)
+        //{
+        //    var json = new JObject(new JProperty("unsyncDrives", unsyncDrives));
+
+        //    var content = new StringContent(json.ToString(), Encoding.UTF8, "application/json");
+        //    var requestUri = GenerateRequestUri(_apiBaseUrl, "drivingsessions");
+        //    var response = await _client.PostAsync(requestUri, content);
+        //    return response.IsSuccessStatusCode;
+        //}
+
+        //public async Task<List<StateReqs>> GetStateReqs()
+        //{
+        //    var requestUri = GenerateRequestUri(_apiBaseUrl, "statereqs");
+        //    var response = await _client.GetAsync(requestUri);
+        //    var json = response.Content.ReadAsStringAsync().Result;
+        //    var stateReqs = JsonConvert.DeserializeObject<List<StateReqs>>(json);
+        //    return stateReqs;
+        //}
+
```

Comment:
I wanted to preserver the workflow of the function of these methods so that when we re write them with our new workflow, I understand how they worked before.